packages/playht/src/grpc-client/client.ts

```diff
@@ -76,8 +86,20 @@ export class Client {
     }
 
     this.leasePromise = this.getLease();
-    this.lease = await this.leasePromise;
-    this.leasePromise = undefined;
+    try {
+      this.lease = await this.leasePromise;
+    } catch (e) {
+      // If this lease fails and we have time left before the deadline, reschedule.
+      const expiresIn = this.lease.expires.getTime() - Date.now();
+      if (expiresIn > 1000 * 15) {
+        clearTimeout(this.leaseTimer);
+        // Try again every 30 seconds until it works or we run out of time.
+        this.leaseTimer = setTimeout(() => this.refreshLease(), Math.min(1000 * 30, expiresIn - 1000 * 10));
+      }
```

Comment:
I disagree.  
1. People shouldn't wait on a retry. We make a blocking call to getLease() when the stream starts, so retries inject latency. Instead, that should die quickly. It's the same reason we return NO_CAPACITY rather than queue the request.
2. Ultimately both types of retry (getLease vs refreshLease) can result in an exception, but trying to use an expired lease will at least generate an error on our side. Right now, this just dies on the client, and we don't know anything about it, other than angry people yelling at me on slack.

The way we built this client makes it very hard for people to catch errors coming from a timer event, so if we're going to quietly refresh the lease in the background, it has to be as automagic as possible. If we leave the fail-fast exception in getLease(), It's at least easier to try/catch stream creation.